js/focuspoint/jquery.focusPoint.js

```diff
@@ -0,0 +1,436 @@
+
+				// Calculate the radius in px of the focusInterface based on width
+				base.focusInterface._radius = base.focusInterface._width / 2;
+
+				// Write offset based on the center point of the focusInterface
+				base.focusInterface._offset.x = base.focusInterface.$el.offset().left + base.focusInterface._radius;
+				base.focusInterface._offset.y = base.focusInterface.$el.offset().top + base.focusInterface._radius;
+
+				// Write position based on the calculation position of focuspoint of the attachment
+				base.focusInterface._position.x = (base.attachment._focusPoint.x / 100) * base.attachment._width;
+				base.focusInterface._position.y = (base.attachment._focusPoint.y / 100) * base.attachment._height;
+			}
+		};
+
+		/**
+		 *  Crop Button functions
+		 */
+		base.cropButton = {
+			$el: false,
+
+			init: function ()
+			{
+				var button = '<button type="button" class="' + css.button.self + ' ' + css.button.disabled + ' crop-attachment ' + css.imageFocus.button + '">' + focusPointL10n.cropButton + '</button>';
+				$(base.el).find('.attachment-actions').append(button);
+
+				base.cropButton.$el = $('.' + css.imageFocus.button);
+
+				//Set action to button for ajax call
+				base.cropButton.$el.on('click', base.sendImageCropDataByAjax);
+			},
+			highlight: function ()
+			{
+				base.cropButton.$el.removeClass(css.button.disabled);
+				base.cropButton.$el.text(focusPointL10n.cropButton);
+				base.cropButton.$el.addClass(css.button.primary);
+			},
+			activate: function ()
+			{
+				base.cropButton.$el.removeClass(css.button.disabled);
+				base.cropButton.$el.removeClass(css.button.primary);
+			},
+			disable: function ()
+			{
+				base.cropButton.$el.removeClass(css.button.primary);
+				base.cropButton.$el.addClass(css.button.disabled);
+			}
+		};
+
+		base.sendImageCropDataByAjax = function ()
+		{
+			var prepData = {
+				id: base.attachment._id,
+				focusPoint: base.attachment._focusPoint
+			};
+
+			$.ajax({
+				type: 'POST',
+				url: ajaxurl,
+				data: {
+					action: 'initialize-crop',
+					attachment: prepData
+				},
+				dataType: 'json',
+				beforeSend: function ()
+				{
+					if (base._ajaxState.crop === true) {
+						return false;
+					}
+
+					base.cropButton.$el.text('Cropping...');
+					base.cropButton.disable();
+					base._ajaxState.crop = true;
+				}
+			}).done(function (data)
+				{
+					if (data.success === true) {
+						base.cropButton.$el.text('Done!');
+					}
+					else if (data.success === false) {
+						base.cropButton.activate();
+						base.cropButton.$el.text('Please try again');
+					}
+
+					base._ajaxState.crop = false;
+				}
+			);
+		};
+
+		// Variables
+
+		base._ajaxState = {
+			crop: false
+		};
+
+		// Run initializer
+		base.init();
+	};
+
+	$.imageFocus.focusPoint.defaultOptions = {
+		myDefaultValue: ""
+	};
+
+	$.fn.imageFocus_focusPoint = function
+		(options)
+	{
+		return this.each(function ()
+		{
+			(new $.imageFocus.focusPoint(this,
```

Comment:
ES lint tells us we need to store this new object in a variable.
https://stackoverflow.com/questions/33287045/eslint-suppress-do-not-use-new-for-side-effects